aiohttp/client_reqrep.py

```diff
@@ -359,9 +359,11 @@ def update_headers(self, headers: Optional[LooseHeaders]) -> None:
         netloc = self.url.host_subcomponent
         assert netloc is not None
         # See https://github.com/aio-libs/aiohttp/issues/3636.
-        netloc = netloc.rstrip(".")
-        if self.url.port is not None and not self.url.is_default_port():
-            netloc += ":" + str(self.url.port)
+        if netloc[-1] == ".":
+            netloc = netloc.rstrip(".")
+        explicit_port = self.url.explicit_port
+        if explicit_port is not None and not self.url.is_default_port():
+            netloc = f"{netloc}:{explicit_port}"
```

Comment:
Isn't the first check redundant?
```suggestion
        if not self.url.is_default_port():
            netloc = f"{netloc}:{self.url.explicit_port}"
```

Reply:
`is_default_port` is never cached in `yarl` (and it could be optimize a bit more) so checking the cached property first is a guard to avoid calling `is_default_port`.  This needs a comment to explain why its first.